tasks/eth/006-MCP-L1/SignFromJsonWithAssertions.s.sol

```diff
@@ -0,0 +1,274 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.15;
+
+import {SignFromJson} from "./SignFromJson.s.sol";
+import {DeployConfig} from "@eth-optimism-bedrock/scripts/DeployConfig.s.sol";
+import {SystemConfig} from "@eth-optimism-bedrock/src/L1/SystemConfig.sol";
+import {Constants, ResourceMetering} from "@eth-optimism-bedrock/src/libraries/Constants.sol";
+import {L1StandardBridge} from "@eth-optimism-bedrock/src/L1/L1StandardBridge.sol";
+import {L2OutputOracle} from "@eth-optimism-bedrock/src/L1/L2OutputOracle.sol";
+import {ProtocolVersion, ProtocolVersions} from "@eth-optimism-bedrock/src/L1/ProtocolVersions.sol";
+import {SuperchainConfig} from "@eth-optimism-bedrock/src/L1/SuperchainConfig.sol";
+import {OptimismPortal} from "@eth-optimism-bedrock/src/L1/OptimismPortal.sol";
+import {L1CrossDomainMessenger} from "@eth-optimism-bedrock/src/L1/L1CrossDomainMessenger.sol";
+import {OptimismMintableERC20Factory} from "@eth-optimism-bedrock/src/universal/OptimismMintableERC20Factory.sol";
+import {L1ERC721Bridge} from "@eth-optimism-bedrock/src/L1/L1ERC721Bridge.sol";
+import {Predeploys} from "@eth-optimism-bedrock/src/libraries/Predeploys.sol";
+import {Types} from "@eth-optimism-bedrock/scripts/Types.sol";
+import {console2 as console} from "forge-std/console2.sol";
+import {stdJson} from "forge-std/StdJson.sol";
+
+contract PostCheck is SignFromJson {
+    /// @dev Verify that the slot number of xDomainMsgSender in lib/optimism/packages/contracts-bedrock/snapshots/storageLayout/L1CrossDomainMessenger.json
+    ///      is 24. If not, update xdmSenderSlotNumber to the correct slot number.
+    uint256 constant xdmSenderSlotNumber = 24;
```

Comment:
I see slot 204, which was what you originally had. When I run `just simulate` it also passes. We should update CI/etc. accordingly to catch this mistake

Reply:
decided to make changes to read `xDomainMsgSender` directly from snapshot!

Reply:
Awesome. Can we still ensure that CI and a local simulation would fail if the wrong `xDomainMsgSender` slot is used? The purpose of this PR is to catch state errors and we clearly missed one, so it makes me nervous that everything passed when we had that bug

Reply:
@0xfuturistic let me know what the fix ended up being here and how I can test out that failures are properly detected

Reply:
ok I added an assert to the forge script that reads the value from the snapshot and checks whether the correct slot was used. this means that if the wrong slot had been used, CI should fail bc the simulation would fail. previously, we didn't have this. do you think this would be good enough? seems that if we wanted to make an individual CI check for this, it'd make more sense to do it in the monorepo, since otherwise that would mean testing a dependency here (which should test that on its own)

Reply:
Sorry, I was mainly referring to the fact that I'd expect this assertion to fail if we have the wrong `xdmSenderSlot` value, and therefore the `just simulate` command should exit with an error.
https://github.com/ethereum-optimism/superchain-ops/blob/6aeba81d59819b0680e649519b68ecb089e41582/tasks/eth/006-MCP-L1/SignFromJson.s.sol#L154

But if I hardcode the wrong slot number into the=is line:
https://github.com/ethereum-optimism/superchain-ops/blob/6aeba81d59819b0680e649519b68ecb089e41582/tasks/eth/006-MCP-L1/SignFromJson.s.sol#L153

And run:
```
just \
  --dotenv-path .env \
  --justfile ../../../single.just \
  simulate \
  council
```

The simulation still succeeds.

This is because we do not call `_postCheck()` anywhere as far as I can tell 😅

This should be called from both public methods in the `script/SignFromJson.s.sol` file

Reply:
looks like when adding `_postCheck() `to function `signJson`, the call fails. for `approveJson` and `runJson` everything goes smoothly (everything in file `script/NestedSignFromJson.s.sol`)

Reply:
Do you need help debugging this?

Reply:
if you have a minute, we could parallelize working on this. do you think you could take a look?

Reply:
I saw the error of `revert: 500` which is from this line:
```solidity
require(configToCheck.scalar() == gasPriceOracleScalar, "500");
```

Logging those two values we see:
```
configToCheck.scalar(): 452312848583266388373324160190187140051835877600158453279134670530344387928
 gasPriceOracleScalar:   0
```

I believe this is because the deploy config's value of 0 is not updated to support the new ecotone encoding. I think it should be safe to hardcode the expected value here to `0x10000000000000000000000000000000000000000000000000c5fc500000558`, which is the hex representation of that large decimal number.

I then commented out this one `require` statement and now it looks like SystemConfig calls are reverting, e.g. `configToCheck.startBlock()` is the next call to fail. We'll need to look into that next

Reply:
could it be that the script is checking against the current version of `SystemConfig`, which doesn't include a method for `startBlock()`?